### DISCUSSION TITLE
Bug: EsLint has introduced a dependency on tsconfig - should be able to lint a non-compiled file without error.

### DISCUSSION THREAD

**Discussion by redevill** [eslint/eslint#18671] [Q&A] [answered]
### Environment

Node version: 18.19.0
npm version: 10.4
Local ESLint version: 8.56
Global ESLint version: 
Operating System: Mac 13.6.6

### What parser are you using?

@angular-eslint/template-parser

### What did you do?

Ran the parser against a specified set of code per eslintrc & eslintignore.

Parsing error: ESLint was configured to run on `<tsconfigRootDir>/envConfigs/eti.ts` using `parserOptions.project`: 
- /<path>/tsconfig.json
- /<path>/server/tsconfig.json
However, none of those TSConfigs include this file. Either:
- Change ESLint's list of included files to not include this file
- Change one of those TSConfigs to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project

### What did you expect to happen?

No errors related to coordination with tsconfig specification.
If so desired, we should be able to run eslint on any files - especially additional files that are not compiled - without error. I can see the need to warn about files that ARE compiled without being linted. 

### What actually happened?

Got an error when linting an alternate environment settings file, that is not directly part of my compilation.

### Link to Minimal Reproducible Example

This is known expected behavior - no repro needed

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
I believe it's because you have enabled some type-checked rules: https://typescript-eslint.io/getting-started/typed-linting.

I'm converting it to a discussion as it's an tseslint-related, which is not maintained by the eslint team.

**Reply by redevill** [reply]
While I am linting typescript files (see plugin list), I am not using the "TypeChecked"  variations you reference.
Been doing the same linting setup for years, with the only change to the linting environment, library updates. Noticed the new error started happening.
My contention is still that we should be able to lint a file regardless of how this file is,  OR is NOT, involved with a compilation.

`      "files": ["*.ts"],
      "parserOptions": {
        "ecmaVersion": 2020,
        "sourceType": "module",
        "project": ["tsconfig.json", "server/tsconfig.json"],
        "createDefaultProgram": true
      },
      "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:@angular-eslint/recommended",
        "plugin:@angular-eslint/template/process-inline-templates",
        "prettier"
      ],`

**Reply by aladdin-add** [reply, marked as answer]
https://typescript-eslint.io/getting-started/typed-linting#how-can-i-disable-type-aware-linting-for-a-subset-of-files

does it work for you?

**Reply by redevill** [reply]
Yes, I did manage to get that to work - Thank you.
